R/defStormsDataset.R

```diff
@@ -241,23 +248,350 @@ checkInputsdefStormsDataset <- function(filename, fields, basin, seasons, unitCo
+  data$isotimes <- data$isotimes[, o]
+  data$longitude <- data$longitude[, o]
+  data$latitude <- data$latitude[, o]
+  data$msw <- data$msw[, o]
+  
+  if ("sshs" %in% names(fields)) {
+    data$sshs <- array(ncdf4::ncvar_get(dataBase, fields["sshs"])[, ind], dim = c(row, len))
+    data$sshs <- data$sshs[, o]
+  }
+  
+  if ("rmw" %in% names(fields)) {
+    if (unitConversion["rmw"] == "nm2km") {
+      data$rmw <- array(nm2km(ncdf4::ncvar_get(dataBase, fields["rmw"])[, ind]), dim = c(row, len))
+    } else {
+      data$rmw <- array(ncdf4::ncvar_get(dataBase, fields["rmw"])[, ind], dim = c(row, len))
+    }
+    data$rmw <- data$rmw[, o]
+  }
+  
+  
+  if ("pressure" %in% names(fields)) {
+    if (unitConversion["pressure"] == "mb2pa") {
+      data$pressure <- array(mb2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["pressure"] == "b2pa") {
+      data$pressure <- array(b2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["pressure"] == "psi2pa") {
+      data$pressure <- array(psi2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["pressure"] == "atm2pa") {
+      data$pressure <- array(atm2pa(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind]), dim = c(row, len))
+    } else {
+      data$pressure <- array(ncdf4::ncvar_get(dataBase, fields["pressure"])[, ind], dim = c(row, len))
+    }
+    data$pressure <- data$pressure[, o]
+  }
+  
+  if ("poci" %in% names(fields)) {
+    if (unitConversion["poci"] == "mb2pa") {
+      data$poci <- array(mb2pa(ncdf4::ncvar_get(dataBase, fields["poci"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["poci"] == "b2pa") {
+      data$poci <- array(b2pa(ncdf4::ncvar_get(dataBase, fields["poci"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["poci"] == "psi2pa") {
+      data$poci <- array(psi2pa(ncdf4::ncvar_get(dataBase, fields["poci"])[, ind]), dim = c(row, len))
+    } else if (unitConversion["poci"] == "atm2pa") {
+      data$poci <- array(atm2pa(ncdf4::ncvar_get(dataBase, fields["poci"])[, ind]), dim = c(row, len))
+    } else {
+      data$poci <- array(ncdf4::ncvar_get(dataBase, fields["poci"])[, ind], dim = c(row, len))
+    }
+    data$poci <- data$poci[, o]
+  }
+  
+  
+  ncdf4::nc_close(dataBase)
+  
+  return(data)
+}
+
+
+
+
+#' Load database when filename is a NetCDF
+#' (CF defStormsDataset for additional informations about parameters)
+#'
+#' @param filename character
+#' @param fields named character vecor
+#' @param basin character
+#' @param seasons numeric vector
+#' @param unitConversion named character vector
+#' @param verbose numeric
+#'
+#' @return list of arrays
+getDataFromCsvFile <- function(filename, fields, basin, seasons, unitConversion, verbose){
+  
+  if (verbose) {
+    cat("=== Loading data  ===\nOpen database... ")
+  }
+  
+  dataBase <- utils::read.csv(filename)
```

Comment:
Should be carrefull here: we should make sure it's ok with different csv separators. At least it shoud work with the default 'read.csv' separator, but also it should through an error message when the separator is different.
This should be tested too.

Reply:
Default separator is `','` . Should we add an input alongside with `filename`  like `sep` which is `NULL` by default but mandatory for csv input with a default value equals to `','` ?

Reply:
Done